src/tags.rs

```diff
@@ -100,7 +165,9 @@ fn read_analysis_string(tag_str: &str, start_tag_pos:usize, version_pos:usize) -
             match part.parse::<f32>() {
                 Ok(val) => {
                     num_read_vals += 1;
-                    vals[index - val_start_pos] = val;
+                    // Format with 15 significant digits to ensure consistency
```

Comment:
So read a string, that is already (well after this change) as 15 digits, create a new string, and parse that? No, silly.